packages/api/cms-api/src/page-tree/page-tree-read-api.ts

```diff
@@ -422,6 +420,19 @@ export function createReadApi(
     };
 }
 
+export function paginateNodes<T>(nodes: T[], options: { offset?: number; limit?: number }) {
```

Comment:
I'd remove the generic and just use `PageTreeNodeInterface` here. For the unit tests you may use `as PageTreeNodeInterface` casts